Content.Shared/Magic/Events/MindSwapSpellEvent.cs

```diff
@@ -0,0 +1,12 @@
+using Content.Shared.Actions;
+
+namespace Content.Shared.Magic.Events;
+
+public sealed partial class MindSwapSpellEvent : EntityTargetActionEvent, ISpeakSpell
+{
+    [DataField]
+    public TimeSpan StunDuration = TimeSpan.FromSeconds(10);
+
+    [DataField]
+    public string? Speech { get; private set; }
+}
```

Comment:
I lied, one more change. We should have separate Stun durations for Target and Caster. Even if they're the same value. This way we could balance at a later time if we wanted one or the other to be longer or shorter.